src/MoBi.Core/Serialization/ORM/Mappers/ProjectMetaDataToProjectMapper.cs

```diff
@@ -119,11 +180,40 @@ private void addEntityToProject(EntityMetaData entity, SerializationContext seri
          else if (deserializedEntity.IsAnImplementationOf<Module>())
             _project.AddModule(deserializedEntity as Module);
 
-         else if (deserializedEntity.IsAnImplementationOf<IWithId>())
-            _project.AddBuildingBlock(deserializedEntity as IBuildingBlock);
+         else if (deserializedEntity.IsAnImplementationOf<IndividualBuildingBlock>())
+            _project.AddIndividualBuildingBlock(deserializedEntity as IndividualBuildingBlock);
 
-         else
+         else if (deserializedEntity.IsAnImplementationOf<ExpressionProfileBuildingBlock>())
```

Comment:
looks like a good use case for pattern matching in C#. This was not available when the code was written but is definitely something wes should use now